### ISSUE TITLE
PLY exporter should handle tex coords more carefully when mNumUVComponents != 2.

### ISSUE THREAD

**Issue by JC3** [assimp/assimp#3829]
Library version: master @ cd42b99
Platform: Windows 10 64-bit
Compiler: MSVC 2019 64-bit

The docs for `aiMesh::mNumUVComponents` state:

> Up to three channels are supported (UVW, for accessing volume or cube maps). If the value is 2 for a given channel n, the component p.z of mTextureCoords[n][p] is set to 0.0f. If the value is 1 for a given channel, p.y is set to 0.0f, too. 

However, both PLY exporters (ASCII and binary) blindly export 2-component texture coordinates regardless of the number of components:

https://github.com/assimp/assimp/blob/74577ae3c79132b2d643ae667a1bd71e0a99f48f/code/AssetLib/Ply/PlyExporter.cpp#L277-L286

https://github.com/assimp/assimp/blob/74577ae3c79132b2d643ae667a1bd71e0a99f48f/code/AssetLib/Ply/PlyExporter.cpp#L340-L347

An additional side effect of this is if a 1- or 3- component texture coord is exported by assimp, then later imported by assimp, it'll lose the component count and just treat it as 2.

The behavior should probably instead be:

- If the tex coord set has 1 component, either don't write 2, or write *coord.x* and *0* instead of *coord.x* and *coord.y*. The exporter should not assume that *coord.y* is set to 0 if the coordinates are marked as containing 1 component. Which of those two options to choose is an open question; writing both is probably better for interoperability although creates the need for a little bit of extra work if the importer is supposed to recognize that only one component is intended. 
- If the tex coord set has 3 components, write all 3.

Note that since PLY doesn't define any texture coordinate properties at all, that leaves assimp *somewhat* free to write 1, 2, or 3 of them. Also, regardless of the behavior, the PLY loaders should be updated to match (related: #3828).

---

Btw, as an aside, I'm assuming `s` and `t` were chosen for Blender compatibility. The Windows 3D Viewer and Paint 3D use `texture_u` and `texture_v` and do not recognize `s` and `t`. MeshLab recognizes both forms. Just, FYI.

**Comment by tellypresence** [comment]
Please add "Export" label to this issue

**Comment by kimkulling** [comment]
Yes, but there is a PR in preparation. See https://github.com/assimp/assimp/pull/6124
